### PR TITLE
Support for Mongoose 5.x + Latest node versions

### PR DESCRIPTION
**Pull request by maxiejbe**
### Mongoose 5.x
I got mongoose 5.x working. Basically I fixed:

**Pluralize**
`var pluralize = require('mongoose/lib/utils').toCollectionName;` was not working as expected. 
I supplied an external pluralize function from https://github.com/blakeembrey/pluralize.

```
var pluralizeFn = require('pluralize');
model.plural(pluralize(model.singular(), pluralizeFn));
```
**Query.cursor() instead of Query.stream()**
```
pipeline(request.baucis.query.stream()); // findOne do not support cursor
```
was used for findOne operation instead of:
```
pipeline(request.baucis.query.cursor());
```
I think that's not possible in mongoose 5 (Cursor works as expected). This change might break backwards compatibility.

### Latest node versions
The main change is to **remove trailer from response header**.

Tested with the following node versions:
- `v9.11.2`
- lts/dubnium: `v10.17.0`
- lts/erbium: `v12.13.0`

Using `express v4.17.1` and `mongoose v5.7.8`

### CODE REVIEW

File: test/controllers.js [hunk 1]
```diff
@@ -593,19 +593,19 @@ describe('Controllers', function () {
       expect(response.statusCode).to.be(201);
       request.post(options, function (err, response, body) {
         if (err) return done(err);
-        expect(response.statusCode).to.be(422);
-        expect(body).to.have.length(1);
-        expect(body[0]).to.have.property('name');
-        expect(body[0]).to.have.property('message', 'Path `name` (Gorgonzola) must be unique.');
-        expect(body[0]).to.have.property('path', 'name');
+        expect(body[0]).to.have.property('message');
```
Comment (swordfish444) [inline test/controllers.js]: @ maxiejbe 

❓ ❔ ❓ ❔ **QUESTION** ❓ ❔ ❓ ❔

Can you explain the reason behind all these spec changes?

File: Model.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 var deco = require('deco');
 var mongoose = require('mongoose');
 var pluralize = require('mongoose/lib/utils').toCollectionName;
+var pluralizeFn = require('pluralize');
```
Comment (swordfish444) [inline Model.js]: @maxiejbe This is odd. Why are we using both the existing pluralize method and this external pluralize module? If there are issues in the existing, why not replace it entirely?

Comment (maxiejbe) [reply Model.js]: Just removed the old pluralize function completely.

File: test/controllers.js (hunk 1 above)
Comment (maxiejbe) [reply test/controllers.js]: Basically the duplicate key message was a bit different than before. I've just adjusted the regex to match the current specs.

**Comment by swordfish444** [comment]
@wprl I've reviewed these changes. It's difficult to rely on CI status when the core baucis branch is in a failed state. I'd suggest emphasising getting that fixed and in the meantime manually run test suite on these changes and if they pass then lets get this merged. Mongoose 5 has been out for quite some time now.

Understanding that you're under a non-compete, if you would like to add me to the maintainers, I can ship a new release.

**Comment by swordfish444** [comment]
@maxiejbe Have you ran the test suite manually, if so, what were the results?

**Comment by maxiejbe** [comment]
> @maxiejbe Have you ran the test suite manually, if so, what were the results?

Yes: 171 passing, 14 pending. 
Node lts/carbon (v8.16.2) + express v4.17.1 + mongoose 5.7.7

**Comment by wprl** [comment]
@maxiejbe @swordfish444 thanks for working on this.  Glad to see it has been useful to you.

Skylar, how's it going?  Long time no see :)  I added you as a collaborator.  I am free of IP restrictions now, but have limited time working full time and a 1 y.o. at home right now.  Let me know if I can be of help and I will do my best.

—wprl

**Comment by swordfish444** [comment]
@wprl Can you merge this and chop a new release?

**Comment by wprl** [comment]
Miraculously I have some time to work on OSS projects for the first time in a long time.  This is now published as v2.0.0.  Thank you @maxiejbe !

**Comment by maxiejbe** [comment]
> Miraculously I have some time to work on OSS projects for the first time in a long time. This is now published as v2.0.0. Thank you @maxiejbe !

Thank you for building Baucis 😄 Hope I can help with anything else in the future.

**Pull request merged**